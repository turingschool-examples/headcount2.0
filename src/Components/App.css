html {
  background-color: #2a393a;
  color: #f4fcfc;
  font-size: 62.5%;
}

h1 {
  color: #f7b488;
  font-size: 3.5em;
  margin: 0 0 1em 0;
  padding: .5em;
  text-align: center;
}