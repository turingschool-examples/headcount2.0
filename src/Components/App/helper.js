import kinderData from '../../data/kindergartners_in_full_day_program.js';


export default class DistrictRepository {
	constructor(stats) {
		this.stats = this.removeDuplicates(kinderData)
		
	}

	removeDuplicates = (kinderData) => {

		return kinderData.reduce((acc, school) => {
			// console.log(this.stats)
			let location = school.Location.toUpperCase();
			let schoolYear = school.TimeFrame
			let roundedData= Math.round(school.Data * 1000) / 1000
			if(!acc[location]) {
				acc[location] = {location, stats: {} 
		}}
			acc[location].stats[schoolYear] = roundedData || 0
			return acc;

		}, {});

	}

	findByName = (districtName = '') => {
		const upperDistrict = districtName.toUpperCase();
		
		if(!this.stats[upperDistrict]) {
			return undefined} 

		if (this.stats[upperDistrict]) {
			return this.stats[upperDistrict]
		}
	}

	findAllMatches = (districtName) => {
		let statsValues = Object.values(this.stats)
		if(!districtName) {
			return statsValues
		}
		return statsValues.reduce((acc, district) => {
		const upperDistrict = districtName.toUpperCase();
			if(district.location.includes(upperDistrict)) {
				acc.push(district)
			}
			return acc
		}, [])

		
	}

	findAverage = () => {
		
	}


}
