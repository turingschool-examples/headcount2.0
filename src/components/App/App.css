/*APP COMPONENT RELATED CSS GOES HERE*/
.App {
  height: 100%;
  width: 100vw;
  margin: 0 auto;
  padding: 0 0 15px 0;
  display: grid;
  grid-template-rows: 25% 25% 1fr;
  grid-template-columns: 1fr;
}