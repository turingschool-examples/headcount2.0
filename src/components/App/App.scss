/*APP COMPONENT RELATED CSS GOES HERE*/
.App {
 background: #272823;
}