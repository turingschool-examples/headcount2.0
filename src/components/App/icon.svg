<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 88.65 88.65"><defs><style>.cls-1{fill:#98eb61;}.cls-2{fill:#fff;}</style></defs><title>icon</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><circle class="cls-1" cx="44.33" cy="44.33" r="44.33"/><path class="cls-2" d="M55.68,37.23a19.31,19.31,0,0,1-3,10.55.69.69,0,0,0,.11,1c.8.79.78.81,0,1.6-1.46,1.46-2.92,2.9-4.36,4.37-.32.32-.53.41-.89,0-.8-.83-.82-.81-1.84-.23a19,19,0,0,1-9,2.64,32.09,32.09,0,0,1-3.58-.18,18.36,18.36,0,0,1-10.24-4.38A19.6,19.6,0,0,1,16.39,43a19.84,19.84,0,0,1-.63-8.71A20.09,20.09,0,0,1,29.68,18,19.27,19.27,0,0,1,37,17.15a19.9,19.9,0,0,1,13.26,6.28,19.32,19.32,0,0,1,4.85,8.78c.4,1.63.52,3.33.76,5Zm-3.59,0A16.48,16.48,0,1,0,35.44,53.71,16.41,16.41,0,0,0,52.09,37.25Z"/><path class="cls-2" d="M66.2,72.71a4.53,4.53,0,0,1-3.47-1.54l-6.42-7-4.61-5c-.6-.66-1.21-1.31-1.84-1.94-.28-.27-.32-.47,0-.75q2.5-2.47,5-5c.25-.25.44-.28.71,0q3.85,3.57,7.72,7.13C64.61,59.78,66,61,67.4,62.25A28.38,28.38,0,0,1,70,64.85,4.46,4.46,0,0,1,70.62,70,4.63,4.63,0,0,1,66.2,72.71Z"/><path class="cls-2" d="M21.41,37.65a14.29,14.29,0,0,1,5-11.31,1,1,0,0,1,1.42,0c.36.42.28.92-.25,1.38a12.74,12.74,0,0,0-3.76,5.55,12.43,12.43,0,0,0,4.92,14.29c.83.54,1.07,1.08.71,1.6s-1,.43-1.74-.1a13.76,13.76,0,0,1-6-8.86C21.56,39.24,21.49,38.24,21.41,37.65Z"/></g></g></svg>