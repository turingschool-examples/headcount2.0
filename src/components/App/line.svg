<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 259.96 3.02"><defs><style>.cls-1{fill:#f9f9f9;}.cls-2{fill:#98eb61;}</style></defs><title>line</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><line class="cls-1" y1="1.51" x2="259.96" y2="1.51"/><path class="cls-2" d="M0,1.51l8.12-.44L16.25.71,32.5,0h195C238.3.52,249.13.9,260,1.51c-10.83.61-21.66,1-32.5,1.5H32.5L16.25,2.31,8.12,1.95Z"/></g></g></svg>