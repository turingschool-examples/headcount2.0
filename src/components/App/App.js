import React, { Component } from 'react';
import CardContainer from '../CardContainer/CardContainer';
import Search from '../Search/Search';
import ComparedDistricts from '../ComparedDistricts/ComparedDistricts';

import DistrictRepository from '../../helper';
import kinderGardenerData from '../../data/kindergartners_in_full_day_program';

import './App.css';

const schoolDistricts = new DistrictRepository(kinderGardenerData);

class App extends Component {
  constructor() {
    super();
    this.state = {
      districts: [],
      comparedDistricts: [
        {
          location: 'hellow',
          stats: {2000: 0.29293}
        }
      ]
    };
  }

  componentDidMount = () => {
    this.getSchoolDistrictData();
  }

  getSchoolDistrictData = () => {
    const districts = Object.values(schoolDistricts.stats);
    this.setState({ districts });
    return districts;
  }

  filterDistricts = (query) => {
    const queriedDistrict = schoolDistricts.findAllMatches(query);

    if (query === '') {
      this.setState({districts: this.getSchoolDistrictData() });
    } else {
      this.setState({districts: queriedDistrict});
    }
  }

  render() {
    const { districts, comparedDistricts } = this.state;

    return (
      <main>
        <ComparedDistricts 
          comparedDistricts={comparedDistricts} 
        />
        <Search
          filterDistricts={this.filterDistricts}
        />
        <CardContainer
          districts={districts}
        />
      </main>
    );
  }
}

export default App;