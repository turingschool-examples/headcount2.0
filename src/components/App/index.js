import React, { Component } from 'react';
import Header from '../Header'
import CardContainer from '../CardContainer'
import Comparator from '../Comparator'
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program'
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      kinder: {},
      comparisons: []
    }
  }

  findAllMatches = (query) => {
    const kinder = new DistrictRepository(kinderData)
    const matchingNames = kinder.findAllMatches(query)
    const allMatches = {}

    matchingNames.forEach(name => {
      const district = kinder.findByName(name)

      allMatches[name] = district
    })

    this.setState({kinder: allMatches})
  }

  adjustComparisons = card => {
    const comparisons = this.state.comparisons

    if (comparisons.length < 2) {
      this.addComparison(comparisons, card)

    } else {
      const oldCard = comparisons.shift()
      this.removeComparison(oldCard)
      this.addComparison(comparisons, card)
    }
  }

  addComparison = (comparisons, card) => {
    const newComparisons = comparisons.push(card)

    this.setState({comparisons: newComparisons})
  }

  removeComparison = oldCard => {

  }

  componentDidMount() {
    const kinder = new DistrictRepository(kinderData)

    this.setState({
      kinder: kinder.stats
    })
  }

  render() {
    const kinder = this.state.kinder
    const comps = this.state.comparisons

    return (
      <div className="App">
        <Header findAllMatches={this.findAllMatches}/>
        { (comps.length > 0 && comps.length > 3) ? <Comparator /> : null}
        <CardContainer
          districts={kinder}
          adjustComparisons={this.adjustComparisons}
        />
      </div>
    );
  }
}

export default App