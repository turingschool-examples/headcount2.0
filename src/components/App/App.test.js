/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App.js';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('has initial state with the data and an empty array of selected cards', () => {
  const wrapper = shallow(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2005: 0.555}, location: 'COLORADO'}};
  
  wrapper.setState({districtData: mockData});
  expect(wrapper.state().districtData).toEqual(mockData);
  expect(wrapper.state().selectedCards).toEqual([]);
});

it("should show a card if its state districtData has data in it", () => {
  const wrapper = shallow(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2005: 0.555}, location: 'COLORADO'}};

  wrapper.setState({districtData: mockData});
  expect(wrapper).toMatchSnapshot();
});

it('should have ONE card in the state selected cards array when a card is selected', () => {
  const wrapper = mount(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  
  wrapper.setState({districtData: mockData});
  expect(wrapper.state().selectedCards).toEqual([]);
  wrapper.find('article').first().simulate('click');
  expect(wrapper.state().selectedCards.length).toEqual(1);
});

it('should have TWO cards in the selectCardArray when two are clicked', () => {
  const wrapper = mount(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  
  wrapper.setState({districtData: mockData});
  wrapper.find('article').first().simulate('click');
  expect(wrapper.state().selectedCards.length).toEqual(1);
  wrapper.find('article').last().simulate('click');
  expect(wrapper.state().selectedCards.length).toEqual(2);
});

it('should have the compare container in the DOM within it when there are two selected cards in the array', () => {
  const wrapper = mount(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  const mockArray = [{ data: { '2004': 0.154, '2014': 0.555 }, location: 'COLORADO' },
    { data: { '2004': 0.222, '2014': 0.777 }, location: 'ACADEMY 20' }];
  
  wrapper.setState({districtData: mockData, selectedCards: mockArray});
  expect(wrapper.find('article.compared-cards-container').length).toEqual(1);
});

it('the handleSubmit function should set the districtData state to the searched item', () => {
  const wrapper = shallow(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  const expected = {"ACADEMY 20": {"data": {"2004": 0.302, "2005": 0.267, "2006": 0.354, "2007": 0.392, "2008": 0.385, "2009": 0.39, "2010": 0.436, "2011": 0.489, "2012": 0.479, "2013": 0.488, "2014": 0.49}, "location": "ACADEMY 20"}};
  const inst = wrapper.instance();

  wrapper.setState({districtData: mockData});
  inst.handleSubmit('academy 20');
  expect(wrapper.state().districtData).toEqual(expected);
});

it('the makeComparison function should show a comparison card when the selectedCards array length is 2', () => {
  const wrapper = mount(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  wrapper.setState({districtData: mockData});
  wrapper.setState({selectedCards: [{data:{2004: 0.154, 2014: 0.665}, location: 'COLORADO'}, {data:{2004: 0.134, 2014: 0.777}, location: 'ACADEMY 20'}]});
  expect(wrapper.state().selectedCards.length).toEqual(2);
  
  const inst = wrapper.instance();
  inst.handleSubmit('academy 20');
  expect(wrapper.find('article.compare-card').length).toEqual(1);
});

it('the deselectCard function should reduce the selectedCards array length by one', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({selectedCards: [{data:{2004: 0.154, 2014: 0.665}, location: 'COLORADO'}, {data:{2004: 0.134, 2014: 0.777}, location: 'ACADEMY 20'}]});
  expect(wrapper.state().selectedCards.length).toEqual(2);
  
  const inst = wrapper.instance();
  inst.deselectCard('COLORADO');

  expect(wrapper.state().selectedCards.length).toEqual(1);
  expect(wrapper.state().selectedCards).toEqual([{data:{2004: 0.134, 2014: 0.777}, location: 'ACADEMY 20'}]);
});

it('the selectCard function should add a card to the selectedCard array', () => {
  const wrapper = shallow(<App />);
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  wrapper.setState({districtData: mockData, selectedCards: []});

  expect(wrapper.state().selectedCards.length).toEqual(0);

  const inst = wrapper.instance();
  inst.selectCard('ACADEMY 20');

  expect(wrapper.state().selectedCards.length).toEqual(1);
  expect(wrapper.state().selectedCards).toEqual([{data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}]);
});

it('the removeComparison function should empty the selectedCards array in state', () => {
  const wrapper = shallow(<App />);
  const mockArray = [{data:{2004: 0.154, 2014: 0.665}, location: 'COLORADO'}, {data:{2004: 0.134, 2014: 0.777}, location: 'ACADEMY 20'}];
  const mockData = {COLORADO: {data:{2004: 0.154, 2014: 0.555}, location: 'COLORADO'}, 'ACADEMY 20': {data:{2004: 0.222, 2014: 0.777}, location: 'ACADEMY 20'}};
  
  wrapper.setState({districtData: mockData, selectedCards: mockArray });
  const inst = wrapper.instance();
  inst.removeComparison();
  expect(wrapper.state().selectedCards.length).toEqual(0);
});



