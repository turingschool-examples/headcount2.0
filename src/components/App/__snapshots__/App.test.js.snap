// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a card if its state districtData has data in it 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          HEADCOUNT
</h1>,
        <Search
          handleSubmit={[Function]}
/>,
        false,
        <CardContainer
          comparedCards={Array []}
          districtData={
                    Object {
                              "COLORADO": Object {
                                "data": Object {
                                  "2004": 0.154,
                                  "2005": 0.555,
                                },
                                "location": "COLORADO",
                              },
                            }
          }
          selectCard={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "HEADCOUNT",
        },
        "ref": null,
        "rendered": "HEADCOUNT",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "comparedCards": Array [],
          "districtData": Object {
            "COLORADO": Object {
              "data": Object {
                "2004": 0.154,
                "2005": 0.555,
              },
              "location": "COLORADO",
            },
          },
          "selectCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            HEADCOUNT
</h1>,
          <Search
            handleSubmit={[Function]}
/>,
          false,
          <CardContainer
            comparedCards={Array []}
            districtData={
                        Object {
                                    "COLORADO": Object {
                                      "data": Object {
                                        "2004": 0.154,
                                        "2005": 0.555,
                                      },
                                      "location": "COLORADO",
                                    },
                                  }
            }
            selectCard={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "HEADCOUNT",
          },
          "ref": null,
          "rendered": "HEADCOUNT",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "comparedCards": Array [],
            "districtData": Object {
              "COLORADO": Object {
                "data": Object {
                  "2004": 0.154,
                  "2005": 0.555,
                },
                "location": "COLORADO",
              },
            },
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
